### ISSUE TITLE
Revoked UFEI Bootloader

### ISSUE THREAD

**Issue by git-pt-andy** [rescuezilla/rescuezilla#525]
When trying to create a Rescuezilla bootable USB drive for Win 11 using the rescuezilla-2.5.1-64bit.noble.iso and RUFUS  I get an error message from RUFUS that says "RUFUS has detected that the ISO you have selected contains a UEFI bootloader that has been revoked and that will produce a "Security Violation" screen, when Secure Boot is enabled on a fully up to date UEFI system".

If I click past this message the USB is not created, it just hangs. I have also tried using BaleEtcher, but that just stops working whether I point to a local file or the URL . Also my system has secure boot disabled.

In short I cannot seem to use the current Rescuezilla ISO.

Suggestions appreciated and Thank you for all your hard work in making Rescuezilla available.

Andy

![rufus error](https://github.com/user-attachments/assets/82e5655b-a566-4cfb-80e6-f1437b9bae25)

**Comment by Walmart2569** [comment]
I have the same issue after downloading a new version of Rufus (specifically 4.6.2208) while trying to re-flash a USB I was using for Ubuntu. This didn't happen on 4.5.

**Comment by kutzki** [comment]
Same thing happening to me. Any solutions?

**Comment by shasheene** [comment]
Thanks for raising this issue. This will require a new version of Rescuezilla to fix.

This article [What the fuck is an SBAT and why does everyone suddenly care](https://mjg59.dreamwidth.org/70348.html) provides a detailed technical explanation of what appears to be the root cause here.

Rescuezilla is built upon the Ubuntu Linux distribution using the Linux kernel and signed bootloader packages it provides, there's been changes in how these are validated for UEFI Secure Boot known as SBAT.

By building a new release (certainly for the new Ubuntu 24.10 Oracular) it will pull in the bootloader packages with this new SBAT level and that should solve the issue.

That said, looking at the 'Ubuntu Changelog' link on the secure boot package [1], I can see an update for Ubuntu 23.10 (Mantic) but one not the older Ubuntu 24.04 (Noble) which is a long-term support release so should also be getting updates.

[1] https://packages.ubuntu.com/noble/shim-signed

**Comment by Cbcary** [comment]
The same thing happened with me, but at least the ISO actually flashed on the USB for me.
![image](https://github.com/user-attachments/assets/c5ba0a18-7543-4dd4-bb97-c37339626314)

**Comment by daisukedaisn** [comment]
I am having the same problem with Rescuezilla 2.5.1 released on 2024-09-08 with this name **rescuezilla-2.5.1-64bit.noble.iso**.(https://github.com/rescuezilla/rescuezilla/releases/download/2.5.1/rescuezilla-2.5.1-64bit.noble.iso). The problem only affects Windows systems or dual boot systems using secure boot. Windows is responsible for the issue. No surprise there.  I am not able to turn off secure boot on my dual boot system. If I could, this version would work. But the Rescuezilla download page gives no warning of this boot loader issue. It seems earlier versions would also have the same issue. The issue comes not from Rescuezilla per se, but from the OS it uses, Ubuntu. Ubuntu has released a version (24.04) with the updated certificate. Ubuntu explains the problem [here](https://discourse.ubuntu.com/t/sbat-self-check-failed-mitigating-the-impact-of-shim-15-7-revocation-on-the-ubuntu-boot-process-for-devices-running-windows/47378).


For Linux users, this is a cascade of an old installation bug which uses the Windows EFI partition instead of the EFI partition created for the Linux install during the installation process. It would be great to fix that problem sometime. It might mean Windows would not run on a dual boot system but Linux could and could later be patched with a new shim.

**Does anyone know when Rescuezilla will release a version using the patched Ubuntu?**

**Comment by shasheene** [comment]
I have tested my solution in my release candidate for Rescuezilla v2.6. The problem appears solved, certainly on the Ubuntu 24.10 (Oracular) with the new Secure Boot shim (after updating the base build environment).

There's still one blocking issue for the actual Rescuezilla v2.6 release (impacting the `partclone-nbd` utility used by the Image Explorer feature).

Until then, here's a preview of: ([Google Drive](https://drive.usercontent.google.com/download?id=1kPd33gPEwYYf3p_amArDhwFgDBygNXyI&export=download)), reuploaded/renamed from the artifacts section here https://github.com/rescuezilla/rescuezilla/actions/runs/12981489271

SHA256SUM:
```
e28e68a3df0738dc7709eeaa0d200f5bf57b67161192c55e59f1a4ccf275f6e6  rescuezilla-2.5.1-47-g78761f24b-64bit.oracular.iso
```

As soon as I figure out how to fix the issue with `partclone-nbd` in the ISO, I'll release the changes as Rescuezilla v2.6.

**Comment by daisukedaisn** [comment]
Thank you so much for the update. I will check daily for release 2.6 and
test it on my system and report back to you.

On Sunday, January 26, 2025, Shasheen Ediriweera ***@***.***>
wrote:
> I have tested my solution in my release candidate for Rescuezilla v2.6.
The problem appears solved, certainly on the Ubuntu 24.10 (Oracular) with
the new Secure Boot shim (after updating the base build environment).
>
> There's still one blocking issue for the actual Rescuezilla v2.6 release
(impacting the partclone-nbd utility used by the Image Explorer feature).
>
> Until then, here's a preview of: (Google Drive), reuploaded/renamed from
the artifacts section here
https://github.com/rescuezilla/rescuezilla/actions/runs/12981489271
>
> SHA256SUM:
>
> e28e68a3df0738dc7709eeaa0d200f5bf57b67161192c55e59f1a4ccf275f6e6
rescuezilla-2.5.1-47-g78761f24b-64bit.oracular.iso
>
> As soon as I figure out how to fix the issue with partclone-nbd in the
ISO, I'll release the changes as Rescuezilla v2.6.
>
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you commented.<
https://ci3.googleusercontent.com/meips/ADKq_Na0HlRjMJxL7x-Hcqx0cy6k45bkANuicGmb_uyNS7yDnDyDL85vCXu49yfwjWfGZhQtPbB2B2YasEu4ZrYxpEA1yT4aXmQpUCEMy7jTvDdSoQCXrnXdE0gKJkIz6Z6MZzl6czDEp97sdpMeHZ17xSkOMOGh_glafSGQDhZYckH584kZuqUfAom45m7XUFNwDn_JtavnnUUiRCzR7D5YEoqPHfu4v5u767WLyF4-_NRdB3q3F-xBT6w=s0-d-e1-ft#https://github.com/notifications/beacon/AYZYX7SI2WCUSFSM2B4RNOT2MW4NFA5CNFSM6AAAAABQV7WR2KWGG33NNVSW45C7OR4XAZNMJFZXG5LFINXW23LFNZ2KUY3PNVWWK3TUL5UWJTU33N3WW.gif>Message
ID: ***@***.***>

**Comment by shasheene** [comment]
The user in #546 said `rescuezilla-2.5.1-47-g78761f24b-64bit.oracular.iso` didn't work for him (through Rufus at least)

**Comment by Kello-hub** [comment]
Hello.

I have the same issue using Rufus 4.6.2208 and Rescuezilla 2.5.1, but found a workaround:
Writing as ISO image fails, but writing as DD image works!

Kind regards

**Comment by Dzomlija** [comment]
I am getting this same error when writing "rescuezilla-2.5.1-64bit.noble.iso" to USB using Rufus 4.6.2208:


![Image](https://github.com/user-attachments/assets/5ad92de4-d94a-4957-8599-027f1d6944cc)
![Image](https://github.com/user-attachments/assets/75be3e82-e017-428e-baa2-0f76b6dd8c75)

**Comment by daisukedaisn** [comment]
Hello Shasheen,
I downloaded the Rescuezilla ISO from google docs using the link you
provided. I verified the SHA256 checksum.
I used Linux Mint's ImageWriter App to load it on a 128 GB SanDisk USB 3.2
Gen 1 Flash Drive.
*The security certificate issue did not arise since the Penguin splash
screen came up. So this is real progress!*
I waited for several minutes while a series of about 5 horizontal dots in
the middle of the screen scrolled by.
But nothing else happened. I do have a lot of memory in my system but I am
not backing up every drive!
I was not sure how long to wait,  but I did not recall this issue in the
past. So I was not able to test the backup function.
I forced a reboot and tried a second time but again it never got past the
flash screen.

Thank you,
Daisuke

**Comment by shasheene** [comment]
Quick heads up that I will not be able to get to the release the upcoming weekend either — I will once again be working on my job’s work for the entire (long) weekend. It has been very busy at work recently — a crunch period with lots of very important deadlines. I will try to release my 2.6 release candidate as an officially labelled pre-release though (which I probably should have done a few weeks ago when I created it).

**Comment by daisukedaisn** [comment]
Hi Shasheen,

I really had to make a backup and so I tried Foxclone. It worked for me.
I will be glad to support your work with Rescuezilla on Patreon and will
try it when you release it officially.
Wishing you good luck with your deadlines. I have been there...

**Comment by shasheene** [comment]
OK, I have had some time to experiment with this issue.

Because Microsoft updates the SBAT generation level data by writing the UEFI BIOS chip on the motherboard, we can view this from Rescuezilla by opening a terminal and running `mokutil --list-sbat-revocations`. On a 2013-era Dell Latitude it printed the following (manually transcribed by me)

```jsx
sbat1,1,2023012900
shim,4
grub,3
grub.debian.4
```

From the Rescuezilla release based on Ubuntu 24.10 (Oracular), I ran `mokutil --set-sbat-policy latest` to update to the latest revocation. After a reboot (Rescuezilla still successfully booted up), I reran the list command and and saw the following (manually transcribed)

```jsx
sbat1,1,2024010900
shim,4
grub,3
grub.debian.4
```

ie, the revocation list is updated.

**After this, I can finally reproduce a security policy violation error, at least on Rescuezilla v2.4.2. Weirdly the existing 2.5.1 continues to work fine.**

Notably the SBAT level `sbat1,1,2024010900` is what Microsoft updated in the August 2024 Windows 11 update that introduced this issue to most users. From  https://infosec.exchange/@wdormann/113006301108129499

<img width="316" alt="Image" src="https://github.com/user-attachments/assets/3c5ce510-72b3-4c3c-b294-da12190bfc18" /> <img width="315" alt="Image" src="https://github.com/user-attachments/assets/9f4a9ae6-4b7f-4080-86bc-0bdf049f9b64" />

Side note: I tested booting an Ubuntu "Oracular" based version of Rescuezilla on an Intel NUC which I deployed a Windows 11 installation (and had it fully update) and had similar results: the SBAT fields seem to be the absolute latest.

As best I can tell right now the release candidate I released above was indeed sufficient to fix the issue. I will do a few more tests using Rufus with my Oracular candidate and the NUC running Windows 11 and the Dell Latitude.

**Comment by shasheene** [comment]
I believe this issue has been resolved in [Rescuezilla v2.6.0](https://github.com/rescuezilla/rescuezilla/releases/latest) based on my investigation in the previous comment.

The Rescuezilla v2.6.0 release based on Ubuntu 24.10 "Oracular" should get passed any UEFI Secure Boot issues as best as I can tell. Any boot issues after the GRUB menu will be unrelated to this, with normal advice about eg, using 'Graphical Fallback Mode' etc if you have problems booting.

**Comment by shasheene** [comment]
I should also note, according to this page (https://github.com/rhboot/shim/blob/main/SbatLevel_Variable.txt) there will be another upcoming revocation for February 2025:
```
Revocations for:
 - Februady 2025 GRUB CVEs

sbat,1,2025021800
shim,4
grub,5
```
I still need to figure out how to increment this on a Secure Boot machine so I can see it in `mokutil --list-sbat-revocations` and test the booting behavior.
